src/components/organisms/account-data/account-data.tsx

```diff
@@ -1,80 +1,94 @@
-import { FC, FormEvent, useEffect } from 'react';
+import { FC, useEffect } from 'react';
+import { FormikHelpers, useFormik } from 'formik';
 import { Card } from 'components/atoms/card';
 import { Heading } from 'components/atoms/typography';
 import { Button } from 'components/molecules/button';
 import { Input } from 'components/molecules/input';
 import { useAppDispatch, useAppSelector } from 'store';
 import { setAccountData } from 'store/profile/slice';
 import { selectProfileAccount } from 'store/profile/selectors';
-import { useFormWithValidation } from 'hooks/use-form-with-validation';
-import styles from './account-data.module.scss';
+import { getValidationSchema } from 'utils/validation';
+import { compareObjectFields } from 'utils/compare-object-fields';
+import { PHONE_MASK, UserData } from 'utils/constants';
 import type { AccountFormData } from './types';
+import styles from './account-data.module.scss';
+
+const schema = getValidationSchema<AccountFormData>(
+  UserData.email,
+  UserData.phone
+);
+
+const initialValues: AccountFormData = {
+  email: '',
+  phone: '',
+};
+
+const fieldsToCompare = ['phone'];
```

Comment:
Если вводим енам с названиями полей для формы, то тогда нужно его везде использовать, чтобы минимизировать ошибки:

```suggestion
const initialValues: AccountFormData = {
  [UserDataFieldNames.Email]: '',
  [UserDataFieldNames.Phone]: '',
};

const fieldsToCompare = [UserDataFieldNames.Phone];
```